src/configuration.jl

```diff
@@ -0,0 +1,333 @@
+"""
+mutable struct Configuration
+
+    Struct that contains everything needed for MC.
+
+    There are three different componenets:
+
+ # Members
+
+ ## Static parameters
+
+ - `seed`: seed to initialize random numebr generator, also serves as the unique pid of the configuration
+
+ - `rng`: a MersenneTwister random number generator, seeded by `seed`
+
+ - `para`: user-defined parameter, set to nothing if not needed
+
+ - `totalStep`: the total number of updates for this configuration
+
+ - `var`: TUPLE of variables, each variable should be derived from the abstract type Variable, see variable.jl for details). Use a tuple rather than a vector improves the performance.
+
+ ## integrand properties
+ 
+- `neighbor::Vector{Tuple{Int, Int}}` : vector of tuples that defines the neighboring integrands. Two neighboring integrands are directly connected in the Markov chain. 
+    e.g., [(1, 2), (2, 3)] means the integrand 1 and 2 are neighbor, and 2 and 3 are neighbor.  
+   The neighbor vector defines a undirected graph showing how the integrands are connected. Please make sure all integrands are connected.
+   By default, we assume the N integrands are in the increase order, meaning the neighbor will be set to [(N+1, 1), (1, 2), (2, 4), ..., (N-1, N)], where the first N entries are for diagram 1, 2, ..., N and the last entry is for the normalization diagram. Only the first diagram is connected to the normalization diagram.
+   Only highly correlated integrands are not highly correlated should be defined as neighbors. Otherwise, most of the updates between the neighboring integrands will be rejected and wasted.
+
+ - `dof::Vector{Vector{Int}}`: degrees of freedom of each integrand, e.g., [[0, 1], [2, 3]] means the first integrand has zero var#1 and one var#2; while the second integrand has two var#1 and 3 var#2. 
+
+ - `observable`: observables that is required to calculate the integrands, will be used in the `measure` function call.
+    It is either an array of any type with the common operations like +-*/^ defined. 
+
+ - `reweight`: reweight factors for each integrands. The reweight factor of the normalization diagram is assumed to be 1. Note that you don't need to explicitly add the normalization diagram. 
+
+ - `visited`: how many times this integrand is visited by the Markov chain.
+
+ ## current MC state
+
+ - `step`: the number of MC updates performed up to now
+
+ - `curr`: the current integrand, initialize with 1
+
+ - `norm`: the index of the normalization diagram. `norm` is larger than the index of any user-defined integrands 
+
+ - `normalization`: the accumulated normalization factor. Physical observable = Configuration.observable/Configuration.normalization.
+
+ - `absWeight`: the abolute weight of the current integrand. User is responsible to initialize it after the contructor is called.
```

Comment:
How? I didn't see any initialization example in tests

Reply:
So it's only responsible for internal user to initialize it in montecarlo() right?